WebDriverAgentLib/Routing/FBElementUtils.m

```diff
@@ -33,20 +50,40 @@ + (NSString *)wdAttributeNameForAttributeName:(NSString *)name
   return matchingTypes.copy;
 }
 
-+ (NSArray<NSString *> *)wdPropertyNames {
-  NSMutableArray *result = [NSMutableArray array];
-  unsigned int propsCount = 0;
-  objc_property_t *properties = protocol_copyPropertyList(objc_getProtocol("FBElement"), &propsCount);
-  for (unsigned int i = 0; i < propsCount; ++i) {
-    objc_property_t property = properties[i];
-    const char *name = property_getName(property);
-    NSString *nsName = [NSString stringWithUTF8String:name];
-    if (nsName && [nsName hasPrefix:WD_PREFIX]) {
-      [result addObject:nsName];
++ (NSDictionary<NSString *, id> *)wdProperties
+{
+  static NSDictionary *propertiesWithGetters;
+  static dispatch_once_t propGettersToken;
+  dispatch_once(&propGettersToken, ^{
+    NSMutableDictionary *result = [NSMutableDictionary new];
```

Comment:
I think it is safer to receive getter names from runtime  - we anyway do it only once, which means it should not slow down the tool. What if somebody add a new getter there or rename an existing one? Then the stuff can be easily broken.